src/solver/solver.go

```diff
@@ -150,18 +150,29 @@ func (r *resolver) resolveTypeNoLateStaticBinding(class, typ string) map[string]
 		className, methodName := meta.UnwrapStaticMethodCall(typ)
 		m, ok := FindMethod(className, methodName)
 		if ok {
-			return r.resolveTypes(className, m.Info.Typ)
+			res = r.resolveTypes(className, m.Info.Typ)
+			if m.TraitName != "" {
+				res = replaceTraitName(res, m.TraitName, m.ClassName)
+			}
+			return res
 		}
 		m, ok = FindMethod(className, "__callStatic")
 		if ok {
+			// Should probably run replaceTraitName here on the result
+			// as well, but I have a good __callStatic trait method
```

Comment:
I assume you meant that you don't have a good example, not that you do?

Reply:
You're right. I fixed a typo.